defaults/main.go

```diff
@@ -0,0 +1,63 @@
+package defaults
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+type Decoder interface {
+	Decode(value any) error
+	DisallowUnknownFields()
+}
+
+// readFromFile reads a configuration file to the value.
+// This function will try to read from given path, if it is empty will try
+// fallback path. If it fails on fallback, it will set value to it's defaults
```

Comment:
A couple of typos
```suggestion
// ReadFrom reads a configuration file to the value.
// This function will try to read from given path, if it is empty will try
// fallback path. If it fails on fallback, it will set value to its defaults
```